src/services/balancer/pools/exits/handlers/swap-exit.handler.spec.ts

```diff
@@ -0,0 +1,44 @@
+import { getBalancer } from '@/dependencies/balancer-sdk';
+import { initBalancerWithDefaultMocks } from '@/dependencies/balancer-sdk.mocks';
+import {
+  defaultBatchSwapResponse,
+  initEthersContractWithDefaultMocks,
+} from '@/dependencies/EthersContract.mocks';
+import { Web3ProviderMock } from '@/dependencies/wallets/wallet-connector-mocks';
+import { vaultService } from '@/services/contracts/vault.service';
+import { GasPriceService } from '@/services/gas-price/gas-price.service';
+import { Pool } from '@/services/pool/types';
+import { aWeightedPool } from '@/__mocks__/weighted-pool';
+import { buildExitParams } from '@tests/unit/builders/join-exit.builders';
+import { DeepMockProxy, mockDeep } from 'vitest-mock-extended';
+import { ref } from 'vue';
+import { ExitType } from './exit-pool.handler';
+import { SwapExitHandler } from './swap-exit.handler';
+
+initBalancerWithDefaultMocks();
+initEthersContractWithDefaultMocks();
+
+const gasPriceServiceMock: DeepMockProxy<GasPriceService> =
+  mockDeep<GasPriceService>();
+
+async function mountSwapExitHandler(pool: Pool) {
+  return new SwapExitHandler(ref(pool), getBalancer(), gasPriceServiceMock);
+}
+
+const exitParams = buildExitParams({
+  bptIn: '0.00000000000000001',
```

Comment:
For swap exits there are two modes:
1. OutGivenIn - where `bptIn` is the relevant input
2. InGivenOut - where `amountsOut` is the the relevant input
Maybe we should have a test for both.